bdk-ffi/src/error.rs

```diff
@@ -0,0 +1,35 @@
+use crate::bitcoin::OutPoint;
+
+use bdk::chain::tx_graph::CalculateFeeError as BdkCalculateFeeError;
+
+use std::fmt;
+
+#[derive(Debug)]
+pub enum CalculateFeeError {
+    MissingTxOut { out_points: Vec<OutPoint> },
+    NegativeFee { fee: i64 },
+}
+
+impl fmt::Display for CalculateFeeError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            CalculateFeeError::MissingTxOut { out_points } => {
+                write!(f, "Missing transaction output: {:?}", out_points)
+            }
+            CalculateFeeError::NegativeFee { fee } => write!(f, "Negative fee value: {}", fee),
+        }
+    }
+}
+
+impl From<BdkCalculateFeeError> for CalculateFeeError {
+    fn from(error: BdkCalculateFeeError) -> Self {
+        match error {
+            BdkCalculateFeeError::MissingTxOut(out_points) => CalculateFeeError::MissingTxOut {
+                out_points: out_points.iter().map(|op| op.into()).collect(),
+            },
+            BdkCalculateFeeError::NegativeFee(fee) => CalculateFeeError::NegativeFee { fee },
+        }
+    }
+}
+
+impl std::error::Error for CalculateFeeError {}
```

Comment:
Implementing the `std::error::Error` trait for `CalculateFeeError` allows error types to integrate with Rust's standard error handling ecosystem